api/src/certification/scoring/domain/models/CertificationAssessmentScoreV3.js

```diff
@@ -108,4 +130,59 @@ const _hasCandidateAnsweredEnoughQuestions = ({ answers }) => {
   return answers.length >= config.v3Certification.scoring.minimumAnswersRequiredToValidateACertification;
 };
 
+const _hasCandidateCompletedTheCertification = ({ answers, maximumAssessmentLength }) => {
+  return answers.length >= maximumAssessmentLength;
+};
+
+const _shouldDowngradeCapacity = ({ maximumAssessmentLength, answers, abortReason }) => {
+  return (
+    _hasCandidateAnsweredEnoughQuestions({ answers }) &&
+    !_hasCandidateCompletedTheCertification({ answers, maximumAssessmentLength }) &&
+    abortReason === ABORT_REASONS.CANDIDATE
+  );
+};
+
+const _downgradeCapacity = ({
+  capacity,
+  allChallenges,
+  allAnswers,
+  flashAlgorithmService,
+  pickAnswerStatusService,
+  pickChallengeService,
+  flashAssessmentAlgorithmConfiguration,
+}) => {
+  const numberOfUnansweredChallenges =
+    flashAssessmentAlgorithmConfiguration.maximumAssessmentLength - allAnswers.length;
+
+  const answerStatusArray = Array.from({ length: numberOfUnansweredChallenges }, () => AnswerStatus.SKIPPED);
+
+  const flashAssessmentAlgorithm = new FlashAssessmentAlgorithm({
+    flashAlgorithmImplementation: flashAlgorithmService,
+    configuration: flashAssessmentAlgorithmConfiguration,
+  });
+
+  const pickAnswerStatus = pickAnswerStatusService.pickAnswerStatusFromArray(answerStatusArray);
+  const pickChallenge = pickChallengeService.chooseNextChallenge();
+
+  const singleMeasureStrategy = new AssessmentSimulatorSingleMeasureStrategy({
+    algorithm: flashAssessmentAlgorithm,
+    challenges: allChallenges,
+    pickChallenge,
+    pickAnswerStatus,
+    initialCapacity: capacity,
+  });
+
+  const getStrategy = () => singleMeasureStrategy;
+
+  const simulator = new AssessmentSimulator({
+    getStrategy,
+  });
```

Comment:
Pour gratter une petite ligne :

 ```suggestion
const simulator = new AssessmentSimulator({
  getStrategy:  () => singleMeasureStrategy,
});
```